File: android/src/main/java/com/capacitorjs/plugins/pushnotifications/NotificationLogItem.java
package com.capacitorjs.plugins.pushnotifications;

import org.json.JSONException;
import org.json.JSONObject;

public class NotificationLogItem {
    String timeStamp;
    String deviceId;
    String notificationLogId;
    String interventionId;
    String origin;

    public NotificationLogItem(String timeStamp, String deviceId, String notificationLogId, String interventionId, String origin) {
        this.timeStamp = timeStamp;
        this.deviceId = deviceId;
        this.notificationLogId = notificationLogId;
        this.interventionId = interventionId;
        this.origin = origin;
    }

    public String toJson() throws JSONException {
        JSONObject json = new JSONObject();
        json.put("timeStamp", timeStamp);
        json.put("deviceId", deviceId);
        json.put("notificationLogId", notificationLogId);
        json.put("interventionId", interventionId);
        json.put("origin", origin);
        return json.toString();
    }
}

File: ios/Sources/PushNotificationsPlugin/NotificationLogItem.swift
import Foundation

public class NotificationLogItem {
    public var timeStamp: Int
    public var deviceId: String
    public var notificationLogId: String
    public var interventionId: String
    public var origin: String
    public var areNotificationsEnabled: Bool? = nil
    public var applicationIsActive: Bool = false

    public init (timeStamp: Int, deviceId: String, notificationLogId: String, interventionId: String,
                origin: String, areNotificationsEnabled: Bool? = nil, applicationIsActive: Bool = false) {
        self.timeStamp = timeStamp
        self.deviceId = deviceId
        self.notificationLogId = notificationLogId
        self.interventionId = interventionId
        self.origin = origin
        self.areNotificationsEnabled = areNotificationsEnabled
        self.applicationIsActive = applicationIsActive
    }

    public func toJson() -> Data? {
        var log = [
            "timeStamp": String(self.timeStamp),
            "deviceId": self.deviceId,
            "notificationLogId": self.notificationLogId,
            "interventionId": self.interventionId,
            "origin": self.origin,
            "applicationIsActive": String(self.applicationIsActive)
        ]

        if (self.areNotificationsEnabled != nil) {
            log["areNotificationsEnabled"] = String(self.areNotificationsEnabled!)
        }

        return try? JSONSerialization.data(withJSONObject: log)
    }
}
